src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -100,19 +100,67 @@ public void ResolveFileProviders(IConfiguration configuration)
                 WebRootFileProvider = new PhysicalFileProvider(WebRootPath);
             }
 
-            if (this.IsDevelopment())
-            {
-                StaticWebAssetsLoader.UseStaticWebAssets(this, configuration);
-            }
         }
 
         public string ApplicationName { get; set; }
         public string EnvironmentName { get; set; }
 
-        public IFileProvider ContentRootFileProvider { get; set; }
-        public string ContentRootPath { get; set; }
 
-        public IFileProvider WebRootFileProvider { get; set; }
-        public string WebRootPath { get; set; }
+        public IFileProvider ContentRootFileProvider
+        {
+            get => _contentRootFileProvider;
+            set => _contentRootFileProvider = value;
+        }
+
+        public IFileProvider WebRootFileProvider
+        {
+            get => _webRootFileProvider;
+            set => _webRootFileProvider = value;
+        }
+
+
+        public string ContentRootPath
+        {
+            get => _contentRootPath;
+            set
+            {
+                _contentRootPath = ResolvePathToRoot(value, AppContext.BaseDirectory);
+                if (Directory.Exists(_contentRootPath))
+                {
+                    _contentRootFileProvider = new PhysicalFileProvider(_contentRootPath);
+                }
+            }
+        }
+
+        public string WebRootPath
+        {
+            get => _webRootPath;
+            set
+            {
+                if (!string.IsNullOrEmpty(value))
```

Comment:
I think we need a test case for 

```
builder.WebHost.UseWebRoot("MyWebRoot");
builder.WebHost.UseWebRoot("");
```

It looks like the WebRoot will stay "MyWebRoot" because of this, but with the generic host it would reset to it's default value of "wwwroot".

Reply:
`CannotUnsetPaths` covers since this scenario since we handle the null/empty string case together. This test case currently assumes that setting a path to null/empty string will no-op so we'll have to change the behavior here to mimic what `ContentRootPath` does (fall back to some default).